### PR TITLE
Remove `windows` from `constant` function

### PR DESCRIPTION
**Pull request by AlbertMitjans**

### CODE REVIEW

File: pennylane/pulse/convenience_functions.py [hunk 1]
```diff
@@ -21,32 +22,45 @@
 except ImportError:
     has_jax = False
 
-
-def constant(windows: List[Tuple[float]] = None):
-    """Returns a callable ``f(p, t)`` that returns ``p`` inside the time
-    windows defined in ``windows``.
+# pylint: disable=unused-argument
+def constant(scalar, time):
+    """Parametrized function with the signature ``f(p, t)`` that is constant in time and returns ``p``.
```
Comment (AlbertMitjans) [inline pennylane/pulse/convenience_functions.py:27]:
```suggestion
    """Parametrized function that is constant in time and returns ``p``.
```

File: pennylane/pulse/convenience_functions.py [hunk 2]
```diff
@@ -21,32 +22,45 @@
...
+    """Parametrized function that is constant in time and returns the given ``scalar``.
 
     Args:
-        windows (Tuple[float, Tuple[float]]): List of tuples containing time windows where
-        ``f(p, t)`` is evaluated. If ``None``, it is always evaluated. Defaults to ``None``.
+        scalar (float): the scalar to be returned
+        time (float): Time. This argument is not used.
+
+    This function is mainly used to build a parametrized hamiltonian that can be differentiated
+    with respect to its constant term.
```
Comment (dwierichs) [inline pennylane/pulse/convenience_functions.py:34]: In order to make clear that constant does not mean untrainable, which would be confusing :)
```suggestion
    This function is mainly used to build a parametrized Hamiltonian that can be differentiated
    with respect to its time-independent term. It is an alias for `lambda scalar, t: scalar`.
```

File: pennylane/pulse/convenience_functions.py [hunk 3]
```diff
@@ -21,32 +22,45 @@
...
 
     The ``constant`` function can be used to create a parametrized hamiltonian
 
-    >>> windows = [(1, 7), (9, 14)]
-    >>> H = qml.pulse.constant(windows) * qml.PauliX(0)
+    >>> H = qml.pulse.constant * qml.PauliX(0)
 
-    When calling the parametrized hamiltonian, ``constant`` will return the input parameter only
-    when the time is inside the given time windows
+    When calling the parametrized hamiltonian, ``constant`` will always return the input parameter
```
Comment (dwierichs) [inline pennylane/pulse/convenience_functions.py:42]:
```suggestion
    When calling the parametrized Hamiltonian, ``constant`` will always return the input parameter
```

File: pennylane/pulse/convenience_functions.py [hunk 4]
```diff
@@ -21,32 +22,45 @@
...
-        ``f(p, t)`` is evaluated. If ``None``, it is always evaluated. Defaults to ``None``.
+        scalar (float): the scalar to be returned
+        time (float): Time. This argument is not used.
+
+    This function is mainly used to build a parametrized hamiltonian that can be differentiated
+    with respect to its constant term.
 
     **Example**
 
     The ``constant`` function can be used to create a parametrized hamiltonian
```
Comment (dwierichs) [inline pennylane/pulse/convenience_functions.py:38]:
```suggestion
    The ``constant`` function can be used to create a parametrized Hamiltonian
```

File: pennylane/pulse/convenience_functions.py [hunk 5]
```diff
@@ -21,32 +22,45 @@
...
 
-    >>> windows = [(1, 7), (9, 14)]
-    >>> H = qml.pulse.constant(windows) * qml.PauliX(0)
+    >>> H = qml.pulse.constant * qml.PauliX(0)
 
-    When calling the parametrized hamiltonian, ``constant`` will return the input parameter only
-    when the time is inside the given time windows
+    When calling the parametrized hamiltonian, ``constant`` will always return the input parameter
 
     >>> params = [5]
```
Comment (dwierichs) [inline pennylane/pulse/convenience_functions.py:44]: Wait, this does not work, does it? `constant` just returns the list, which is not a scalar... Or does this work anyways, unwrapping the single parameter in `[5]`?

**Review by dwierichs — COMMENTED** [review]
Looks good, up to capitalization of Hamiltonian, a tiny suggestion, and a question regarding the example of `scalar=[5]`.

File: pennylane/pulse/convenience_functions.py (hunk 5 above)
Comment (AlbertMitjans) [reply pennylane/pulse/convenience_functions.py:44]: The input to the `ParametrizedHamiltonian` class should always be a `list` containing the parameters for each callable.

Comment (AlbertMitjans) [reply pennylane/pulse/convenience_functions.py:44]: Maybe we should make this clearer in the documentation

Comment (dwierichs) [reply pennylane/pulse/convenience_functions.py:44]: Ah, yes of course. Sorry I was thinking too much about the single function..
Being more clear never hurts but I think this one is on me :D

**Review by dwierichs — APPROVED** [review]
Looks good to me, thanks @AlbertMitjans :)

**Pull request merged**